packages/components/src/palette-edit/test/index.tsx

```diff
@@ -82,18 +83,282 @@ describe( 'getNameForPosition', () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				...colors,
+				{
+					color: '#000',
+					name: 'Color 1',
+					slug: 'color-1',
+				},
+			] );
+		} );
+	} );
+
+	it( 'can not add a new palette', () => {
+		render( <PaletteEdit { ...defaultProps } canOnlyChangeValues /> );
+
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Add color',
+			} )
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'can delete palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Show details',
+			} )
+		);
+		await user.click( screen.getByText( 'Primary' ) );
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Remove color',
+			} )
+		);
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				colors[ 1 ],
+			] );
+		} );
+	} );
+
+	it( 'can update palette name', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Show details',
+			} )
+		);
+		await user.click( screen.getByText( 'Primary' ) );
+		await user.click(
+			screen.getByRole( 'textbox', {
+				name: 'Color name',
+			} )
+		);
+		const nameInput = screen.getByRole( 'textbox', {
+			name: 'Color name',
+		} );
+		await user.clear( nameInput );
+		await user.type( nameInput, 'Primary Updated' );
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				{
+					...colors[ 0 ],
+					name: 'Primary Updated',
+					slug: 'primary-updated',
+				},
+				colors[ 1 ],
+			] );
+		} );
+	} );
+
+	it( 'can update color palette value', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click( screen.getByLabelText( 'Color: Primary' ) );
+		const hexInput = screen.getByRole( 'textbox', {
+			name: 'Hex color',
+		} );
+		await user.clear( hexInput );
+		await user.type( hexInput, '000000' );
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				{
+					...colors[ 0 ],
+					color: '#000000',
+				},
+				colors[ 1 ],
+			] );
+		} );
+	} );
+
+	it( 'can update gradient palette value', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } gradients={ gradients } /> );
+
+		await user.click( screen.getByLabelText( 'Gradient: Pale ocean' ) );
+
+		const typeSelectElement = screen.getByRole( 'combobox', {
+			name: 'Type',
+		} );
+		await user.selectOptions( typeSelectElement, 'radial-gradient' );
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenLastCalledWith( [
```

Comment:
Somehow I had to use `toHaveBeenLastCalledWith` here. Where I used `toHaveBeenCalledTimes`, received number of calls was 5, so maybe that has something to do with it.